pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -327,7 +327,7 @@ func (rc *reconciler) reportMultiAttachError(volumeToAttach cache.VolumeToAttach
 		// We did not find any pods that requests the volume. The pod must have been deleted already.
 		simpleMsg, _ := volumeToAttach.GenerateMsg("Multi-Attach error", "Volume is already exclusively attached to one node and can't be attached to another")
 		for _, pod := range volumeToAttach.ScheduledPods {
-			rc.recorder.Eventf(pod, v1.EventTypeWarning, kevents.FailedAttachVolume, simpleMsg)
+			rc.recorder.Eventf(pod, nil, v1.EventTypeWarning, kevents.FailedAttachVolume, "Attaching Volume", simpleMsg)
```

Comment:
I think the existing Pod event for failed volume attach makes sense because we attach volumes as a result of a Pod being scheduled, and:
1) It may not be obvious to someone to check PVC events if their Pod fails to come up
2) We also have the legacy inline volumes ability that doesn't have a separate PVC object.

It would be nice to have PVC as the secondary object here but I think we can consider that as a followup. The PVC object is not easily accessible at this layer; you need to step down a few structs and then do an object lookup.

Reply:
@wojtek-t do you have a good example of what a good "action" is? I am having trouble understanding the purpose of the field from the interface comments.

Reply:
> It would be nice to have PVC as the secondary object here but I think we can consider that as a followup. 

Maybe I should have been more clear:
- I agree that the primary object should be Pod as is (the primary is called "regarding")
- What I proposed is that the secondary object should be volume ("related" can be though of secondary) - I agree with you that it should be PVC, not PV.

Technically - I would like to avoid followup, because events can be to some extent thought as API, so I would like to avoid frequent changes there. But if this is hard for some reasons to get it here, I can probably live without it :)


> @wojtek-t do you have a good example of what a good "action" is? I am having trouble understanding the purpose of the field from the interface comments.

Yeah - we should clarify this. The good examples are in scheduler, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L385
[there are couple more in this file and in other file in pkg/scheduler]

The convention we used there was:
- reason is "what exactly happend" - in this case: "FailedScheduling"
- action is "what we were trying to do" - in this case ", "Scheduling"
"